gpio/__init__.py

```diff
@@ -0,0 +1,280 @@
+# -*- coding: utf-8 -*-
+__version__ = '1.0.0'
+
+from threading import Lock
+import os
+
+
+_export_lock = Lock()
+_open_pins = {}
+
+
+GPIO_ROOT = '/sys/class/gpio'
+GPIO_EXPORT = os.path.join(GPIO_ROOT, 'export')
+GPIO_UNEXPORT = os.path.join(GPIO_ROOT, 'unexport')
+FMODE = 'w+'  # w+ overwrites and truncates existing files
+IN, OUT = 'in', 'out'
+LOW, HIGH = 0, 1
+BCM = None  # Exists for RPi.GPIO compatibility
+
+
+class GPIOPin(object):
+    """Handle pin state.
+
+    Keeps track of file nodes and functions related to a pin.
+
+    Args:
+        pin (int): the GPIO pin to set up
+    """
+    def __init__(self, pin, mode=None, initial=LOW, active_low=None):
+        existing = _open_pins.get(pin)
+        if existing:
+            existing.cleanup()
+            del existing
+
+        self.pin = str(pin)
+        self.value = None
+
+        self.root = os.path.join(GPIO_ROOT, 'gpio{0}'.format(pin))
+
+        if not os.path.exists(self.root):
+            with _export_lock:
+                with open(GPIO_EXPORT, 'w') as f:
+                    f.write(self.pin)
+                    f.flush()
+
+        # Using unbuffered binary IO is ~ 3x faster than text
+        self.value = open(os.path.join(self.root, 'value'), 'wb+', buffering=0)
+
+        if mode is not None:
+            self.set_direction(mode)
+
+        if active_low is not None:
+            self.set_active_low(active_low)
+
+        if mode == OUT:
+            self.write(initial)
+
+        # Add class to open pins
+        _open_pins[pin] = self
+
+    def get_direction(self):
+        '''Get the direction of pin
+
+        Returns:
+            str: "in" or "out"
+        '''
+        with open(os.path.join(self.root, 'direction'), FMODE) as f:
+            return f.read().strip()
+
+    def set_direction(self, mode):
+        '''Set the direction of pin
+
+        Args:
+            mode (str): "in" or "out"

```

Comment:
add "low" "high"

Consider using the constants instead of the strings for the docs.

Reply:
"low" / "high" are technically pin *states* while "in" / "out" are directions/modes. Mode should probably be "direction".